ImP/imageProcessing/arucoMarkerDetectionAlgorithm.md

```diff
@@ -61,7 +51,144 @@ static void _detectCandidates(InputArray _image, vector< vector< Point2f > >& ca
    vector< vector< Point > > contours;
    /// 2. DETECT FIRST SET OF CANDIDATES
    _detectInitialCandidates(grey, candidates, contours, _params);
+```
+### _detectInitialCandidates
+```
+/**
+* @brief Initial steps on finding square candidates
+*/
+static void _detectInitialCandidates(const Mat &grey, vector< vector< Point2f > > &candidates,
+                                    vector< vector< Point > > &contours,
+                                    const Ptr<DetectorParameters> &params) {
+
+   CV_Assert(params->adaptiveThreshWinSizeMin >= 3 && params->adaptiveThreshWinSizeMax >= 3);
+   CV_Assert(params->adaptiveThreshWinSizeMax >= params->adaptiveThreshWinSizeMin);
+   CV_Assert(params->adaptiveThreshWinSizeStep > 0);
+
+   
+```
+More information about scale space, click [here](https://en.wikipedia.org/wiki/Scale_space)
+
+```
+   // number of window sizes (scales) to apply adaptive thresholding
+   int nScales =  (params->adaptiveThreshWinSizeMax - params->adaptiveThreshWinSizeMin) /
+                     params->adaptiveThreshWinSizeStep + 1;
+
+   vector< vector< vector< Point2f > > > candidatesArrays((size_t) nScales);
+   vector< vector< vector< Point > > > contoursArrays((size_t) nScales);
+```
+[Threshold](https://en.wikipedia.org/wiki/Thresholding_(image_processing)) the image at different scales to find different markers.
```

Comment:
Can we comment in the algorithms being used or link the Math(if we write it ourselves)?